echopype/echodata/combine.py

```diff
@@ -435,14 +446,296 @@ def _check_echodata_channels(
     return channel_selection
 
 
+def _check_ascending_ds_times(ds_list: List[xr.Dataset], ed_group: str) -> None:
+    """
+    A minimal check that the first time value of each Dataset is less than
+    the first time value of the subsequent Dataset. If each first time value
+    is NaT, then this check is skipped.
+
+    Parameters
+    ----------
+    ds_list: list of xr.Dataset
+        List of Datasets to be combined
+    ed_group: str
+        The name of the ``EchoData`` group being combined
+
+    Returns
+    -------
+    None
+
+
+    Raises
+    ------
+    RuntimeError
+        If the timeX dimension is not in ascending order
+        for the specified echodata group
+    """
+
+    # get all time dimensions of the input Datasets
+    ed_time_dim = set(ds_list[0].dims).intersection(POSSIBLE_TIME_DIMS)
+
+    for time in ed_time_dim:
+        # gather the first time of each Dataset
+        first_times = []
+        for ds in ds_list:
+            times = ds[time].values
+            if isinstance(times, np.ndarray):
+                # store first time if we have an array
+                first_times.append(times[0])
+            else:
+                # store first time if we have a single value
+                first_times.append(times)
+
+        first_times = np.array(first_times)
+
+        # skip check if all first times are NaT
+        if not np.isnan(first_times).all():
+            is_descending = (np.diff(first_times) < np.timedelta64(0, "ns")).any()
+
+            if is_descending:
+                raise RuntimeError(
+                    f"The coordinate {time} is not in ascending order for "
+                    f"group {ed_group}, combine cannot be used!"
+                )
+
+
+def _check_filter_params(
```

Comment:
Maybe change this to `_check_no_append_vendor_params` or something similar since the concept of filter does not exist for sonar model other than EK80, but there are still vendor-specific params that are not varying across the appending dimensions (e.g., `ping_time` in AZFP)